### PR TITLE
Allow for python sets to be counted as sets

### PR DESCRIPTION
**Pull request by thecaligarmo**
Fixes Issue #32265 

### :memo: Checklist

- [x] The title is concise and informative.
- [x] The description explains in detail what this PR is about.
- [x] I have linked a relevant issue or discussion.
- [x] I have created tests covering the changes.
- [x] I have updated the documentation and checked the documentation preview.

### CODE REVIEW

File: src/sage/combinat/subset.py [hunk 1]
```diff
@@ -321,9 +321,14 @@ def __contains__(self, value):
...
             False
+            sage: {1, 2} in S
+            True
         """
         if value not in Sets():
-            return False
+            if isinstance(value, set):
+                value = Set(value)
+            else:
+                return False
```
Comment (tscrim) [inline src/sage/combinat/subset.py]:
```suggestion
            return False
```

File: src/sage/combinat/subset.py [hunk 2]
```diff
@@ -321,9 +321,14 @@ def __contains__(self, value):
             True
             sage: 2 in S
             False
+            sage: {1, 2} in S
+            True
         """
         if value not in Sets():
```
Comment (tscrim) [inline src/sage/combinat/subset.py]:
```suggestion
        if value not in Sets() and not isinstance(value, (set,frozenset)):
```

**Review by tscrim — APPROVED** [review]
I think this is fine since all it needs to do is iterate over the object. So basically we add an additional check for `set` (and `frozenset`).

**Comment by tscrim** [comment]
Once this change is done, then this is positive review.

**Comment by thecaligarmo** [comment]
> I think this is fine since all it needs to do is iterate over the object. So basically we add an additional check for `set` (and `frozenset`).

That looks good for me. I've used that instead.

**Comment by thecaligarmo** [comment]
Just ran through and tested everything on my machine and it's all running good. I fixed one style issue (space after comma) and it looks good on my end now.

**Review by tscrim — APPROVED** [review]
Thank you. LGTM.

**Pull request merged**